### PR TITLE
Make genai use common

### PR DESCRIPTION
**Pull request by rlazo**

### CODE REVIEW

**Review by davidmotson — APPROVED** [review]

File: common/src/test/java/com/google/ai/client/generativeai/common/StreamingSnapshotTests.kt [hunk 1]
```diff
@@ -22,9 +22,13 @@ import com.google.ai.client.generativeai.common.shared.HarmCategory
 import com.google.ai.client.generativeai.common.shared.TextPart
 import com.google.ai.client.generativeai.common.util.goldenStreamingFile
 import io.kotest.assertions.throwables.shouldThrow
+<<<<<<< HEAD
```
Comment (davidmotson) [inline common/src/test/java/com/google/ai/client/generativeai/common/StreamingSnapshotTests.kt:25]: unintended merge conflict?

File: generativeai/src/main/java/com/google/ai/client/generativeai/GenerativeModel.kt [hunk 2]
```diff
@@ -71,7 +71,14 @@ internal constructor(
     generationConfig,
     safetySettings,
     requestOptions,
-    APIController(apiKey, modelName, requestOptions)
+    APIController(
+      apiKey,
+      modelName,
+      com.google.ai.client.generativeai.common.RequestOptions(
```
Comment (davidmotson) [inline generativeai/src/main/java/com/google/ai/client/generativeai/GenerativeModel.kt:77]: I think requestoption here should be in the internal SDK

File: generativeai/src/test/java/com/google/ai/client/generativeai/GenerativeModelTests.kt [hunk 3]
```diff
@@ -16,91 +16,13 @@
...
-    @Parameterized.Parameters
-    fun data() =
-      listOf(
-        arrayOf("gemini-pro", "models/gemini-pro"),
-        arrayOf("x/gemini-pro", "x/gemini-pro"),
-        arrayOf("models/gemini-pro", "models/gemini-pro"),
-        arrayOf("/modelname", "/modelname"),
-        arrayOf("modifiedNaming/mymodel", "modifiedNaming/mymodel"),
-      )
+  fun `TODO`() {
```
Comment (daymxn) [inline generativeai/src/test/java/com/google/ai/client/generativeai/GenerativeModelTests.kt:25]: def should be done before we merge this tbh

Comment (rlazo) [reply generativeai/src/test/java/com/google/ai/client/generativeai/GenerativeModelTests.kt]: I've added a very basic set of tests to unblock us.

There's some repeated code for testing between common and genai, this is expected for now. A follow up PR will contain a "common-test" module to make it easier to share code among both SDKs.

File: common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt [hunk 4]
```diff
@@ -64,12 +64,20 @@ val JSON = Json {
  * @property apiVersion the endpoint version to communicate with.
  * @property timeout the maximum amount of time for a request to take in the initial exchange.
```
Comment (daymxn) [inline common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt:65]: These are out of date now, as they were moved to `RequestOptions`

**Review by daymxn — APPROVED** [review]

**Pull request merged**